src/browser/BrowserAction.cpp

```diff
@@ -127,171 +136,150 @@ QJsonObject BrowserAction::handleChangePublicKeys(const QJsonObject& json, const
-    return buildResponse(action, message, newNonce);
+    return buildResponse(action, message, browserRequest.incrementedNonce);
 }
 
 QJsonObject BrowserAction::handleGetLogins(const QJsonObject& json, const QString& action)
 {
-    const QString hash = browserService()->getDatabaseHash();
-    const QString nonce = json.value("nonce").toString();
-    const auto incrementedNonce = browserMessageBuilder()->incrementNonce(nonce);
-    const QString encrypted = json.value("message").toString();
-
     if (!m_associated) {
         return getErrorReply(action, ERROR_KEEPASS_ASSOCIATION_FAILED);
     }
 
-    const QJsonObject decrypted = decryptMessage(encrypted, nonce);
-    if (decrypted.isEmpty()) {
+    const auto browserRequest = decodeRequest(json);
+    if (browserRequest.isEmpty()) {
         return getErrorReply(action, ERROR_KEEPASS_CANNOT_DECRYPT_MESSAGE);
     }
 
-    const QString siteUrl = decrypted.value("url").toString();
+    const auto siteUrl = browserRequest.getString("url");
     if (siteUrl.isEmpty()) {
         return getErrorReply(action, ERROR_KEEPASS_NO_URL_PROVIDED);
     }
 
-    const QJsonArray keys = decrypted.value("keys").toArray();
+    const auto keys = browserRequest.getArray("keys");
 
     StringPairList keyList;
-    for (const QJsonValue val : keys) {
-        const QJsonObject keyObject = val.toObject();
+    for (const auto val : keys) {
+        const auto keyObject = val.toObject();
         keyList.push_back(qMakePair(keyObject.value("id").toString(), keyObject.value("key").toString()));
     }
 
-    const QString id = decrypted.value("id").toString();
-    const QString formUrl = decrypted.value("submitUrl").toString();
-    const QString auth = decrypted.value("httpAuth").toString();
+    const auto id = browserRequest.getString("id");
+    const auto formUrl = browserRequest.getString("submitUrl");
+    const auto auth = browserRequest.getString("httpAuth");
     const bool httpAuth = auth.compare(TRUE_STR) == 0;
 
-    const QJsonArray users = browserService()->findMatchingEntries(id, siteUrl, formUrl, "", keyList, httpAuth);
+    EntryParameters entryParameters;
+    entryParameters.dbid = id;
+    entryParameters.hash = browserRequest.hash, entryParameters.siteUrl = siteUrl;
+    entryParameters.formUrl = formUrl;
+
+    const QJsonArray users = browserService()->findEntries(entryParameters, keyList, httpAuth);
     if (users.isEmpty()) {
         return getErrorReply(action, ERROR_KEEPASS_NO_LOGINS_FOUND);
     }
 
-    QJsonObject message = browserMessageBuilder()->buildMessage(incrementedNonce);
+    auto message = browserMessageBuilder()->buildMessage(browserRequest.incrementedNonce);
     message["count"] = users.count();
     message["entries"] = users;
-    message["hash"] = hash;
+    message["hash"] = browserRequest.hash;
     message["id"] = id;
 
-    return buildResponse(action, message, incrementedNonce);
+    return buildResponse(action, message, browserRequest.incrementedNonce);
```

Comment:
This pattern is all over the new code base: buildMessage, maybe add message params, build response passing in message. I feel like we can simplify this greatly:

buildResponse(action, nonce, QMap<QString, QVariant> params);

then you call it like:

buildResponse(action, nonce, parammap);

use `message[parammap.key] = QJsonValue::fromVariant(parammap.value);`

Reply:
This should work fine:
```
typedef QMap<QString, QVariant> Parameters;

...

const Parameters params {
    { "name", newGroup["name"] },
    { "uuid", newGroup["uuid"] }
};
return buildResponse(action, browserRequest.incrementedNonce, params);

...

Parameters::const_iterator i;
for (i = params.constBegin(); i != params.constEnd(); ++i) {
    message[i.key()] = QJsonValue::fromVariant(i.value());
}
```

Reply:
Yah that is great! Chops down some boiler plate code